packages/amplify-provider-awscloudformation/src/network/stack.ts

```diff
@@ -0,0 +1,196 @@
+import * as apigw2 from '@aws-cdk/aws-apigatewayv2';
+import * as ec2 from '@aws-cdk/aws-ec2';
+import * as ecs from '@aws-cdk/aws-ecs';
+import * as discovery from '@aws-cdk/aws-servicediscovery';
+import * as cdk from '@aws-cdk/core';
+import { prepareApp } from '@aws-cdk/core/lib/private/prepare-app';
+
+type NetworkStackProps = {
+  stackName: string;
+  vpcName: string;
+  vpcId: string;
+  internetGatewayId: string;
+  subnetCidrs: ReadonlyMap<string, string>;
+};
+
+export const NETWORK_STACK_LOGICAL_ID = 'NetworkStack';
+
+export class NetworkStack extends cdk.Stack {
+  constructor(scope: cdk.Construct, id: string, props: NetworkStackProps) {
+    super(scope, id);
+
+    const { stackName, vpcId = '', vpcName, internetGatewayId = '', subnetCidrs = new Map<string, string>() } = props;
+
+    const { outputVpc, outputIgw, vpcCidrBlock } = createVpc(this, vpcId, vpcName, internetGatewayId);
+
+    createAmplifyEnv(this, stackName, outputVpc as any, vpcCidrBlock, outputIgw as any, subnetCidrs);
+  }
+
+  toCloudFormation() {
+    prepareApp(this);
+
+    const cfn = this._toCloudFormation();
+
+    return cfn;
+  }
+}
+
+function createVpc(scope: cdk.Construct, vpcId: string, vpcName: string, internetGatewayId: string) {
+  const vpcCidrBlock = '10.0.0.0/16';
+  const condition = new cdk.CfnCondition(scope, 'UseNewVpcCondition', {
+    expression: cdk.Fn.conditionAnd(cdk.Fn.conditionEquals(vpcId, ''), cdk.Fn.conditionEquals(internetGatewayId, '')),
+  });
+
+  const vpc = new ec2.Vpc(scope, vpcName, {
+    cidr: vpcCidrBlock,
+    subnetConfiguration: [],
+  });
+  (vpc.node.defaultChild as ec2.CfnVPC).tags.setTag('Name', vpcName);
```

Comment:
Please verify that the tags will be present after CLI applies `tags.json` tags to the resources.